pkg/controller/nodemaintenance/check_quorum_size_test.go

```diff
@@ -0,0 +1,171 @@
+package nodemaintenance
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sfakeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+func areErrorsEqual(err1, err2 error) bool {
+	return (err1 != nil && err2 != nil && err1.Error() == err2.Error()) ||
+		(err1 == nil && err2 == nil)
+}
+
+var _ = Describe("checkQuorumValidity", func() {
+	type TestCaseDefinition struct {
+		info            string
+		currentHealthy  int32
+		requiredHealthy int32
+		disruptedPods   map[string]metav1.Time
+		isValidQuorum   bool
+	}
+	It("check fallback option of quorum check", func() {
+		testCases := []TestCaseDefinition{
+			{
+				info:            "quorum broken",
+				currentHealthy:  2,
+				requiredHealthy: 3,
+				isValidQuorum:   false,
+			},
+			{
+
+				info:            "quorum valid",
+				currentHealthy:  3,
+				requiredHealthy: 2,
+				isValidQuorum:   true,
+			},
+			{
+				info:            "quorum not broken with dirsupted pods",
```

Comment:
A disrupted pod effectively just subtracts 1 from the currentHealthy count.
So I don't think this is testing the right condition.

I believe you want `requiredHealthy: 4` and look for `isValidQuorum: false`

Reply:
UpAfterNMO := requiredHealthy(4) - len(disruptedPods) - 1 /* the node to be taken offline */ == 2.

UpAfterNMO (2) must be >= requiredHealthy (2) , therefore the test must pass.

Reply:
I believe we discussed previously, that the test will pass even if `disruptedPods` is `nil`.  
That makes it a bad test.

If you set `requiredHealthy: 4` and look for `isValidQuorum: false`, then you validate that `disruptedPods` are being correctly taken into account.

Reply:
added this test case "